### PR TITLE
upgrade to TS 5.6.2

### PR DESCRIPTION
**Pull request by kamilafsar**
There are some changes which are not compatible with older versions for TS.

### CODE REVIEW

**Review by nloomans — CHANGES_REQUESTED** [review]
This introduces peer-dependency issues:

```
lerna ERR! npm install exited 1 in '@phero/dev'
lerna ERR! npm install stderr:
npm ERR! code ERESOLVE
npm ERR! ERESOLVE could not resolve
npm ERR! 
npm ERR! While resolving: ts-jest@27.1.3
npm ERR! Found: typescript@5.6.2
npm ERR! node_modules/typescript
npm ERR!   dev typescript@"^5.6.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer typescript@">=3.8 <5.0" from ts-jest@27.1.3
npm ERR! node_modules/ts-jest
npm ERR!   dev ts-jest@"^27.0.7" from the root project
npm ERR! 
npm ERR! Conflicting peer dependency: typescript@4.9.5
npm ERR! node_modules/typescript
npm ERR!   peer typescript@">=3.8 <5.0" from ts-jest@27.1.3
npm ERR!   node_modules/ts-jest
npm ERR!     dev ts-jest@"^27.0.7" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

This can be reproduced by running `npx lerna bootstrap`. It appears that a few other dependencies need to be upgraded as well for us to be able to upgrade TypeScript.

**Comment by kamilafsar** [comment]
Alright, good catch @nloomans ! Will fix

File: packages/core/src/parsePheroApp/parseFunctionDefinition.ts [hunk 1]
```diff
@@ -36,10 +36,12 @@ function parseFunctionName(
     | ts.Identifier
     | ts.StringLiteral
     | ts.NumericLiteral
+    | ts.BigIntLiteral
     | ts.ComputedPropertyName
     | ts.PrivateIdentifier
     | ts.ObjectBindingPattern
-    | ts.ArrayBindingPattern,
+    | ts.ArrayBindingPattern
+    | ts.NoSubstitutionTemplateLiteral,
```
Comment (nloomans) [inline packages/core/src/parsePheroApp/parseFunctionDefinition.ts:44]: Question: should these new types be checked as well? I tried to create a function with a `BigIntLiteral` as name, but I didn't manage to without using an `ObjectBindingPattern`, which we already test for. But maybe there is a way I don't know about.

**Review by nloomans — APPROVED** [review]
Looks good! Please do look at the thread I created before merging though.

File: packages/core/src/parsePheroApp/parseFunctionDefinition.ts (hunk 1 above)
Comment (kamilafsar) [reply packages/core/src/parsePheroApp/parseFunctionDefinition.ts:44]: Good catch!

**Pull request merged**